src/components/Questionnaire/CustomAutocomplete.tsx

```diff
@@ -0,0 +1,298 @@
+    newValue: AutocompleteValue<string[], false, false, false>,
+    reason: AutocompleteChangeReason
+  ): void => {
+    if (typeof onChange === "function") {
+      onChange(event, newValue, reason);
+    }
+
+    processValue(newValue);
+    setError(false);
+  };
+
+  const handleInputBlur = () => {
+    setHasFocus(false);
+    sortOptions(true);
+  };
+
+  const sortOptions = (force = false) => {
+    if (hasFocus && !force) {
+      return;
+    }
+
+    const selectedOptions = val
+      .filter((v) => options.includes(v))
+      .sort((a, b) => a.localeCompare(b));
+    const unselectedOptions = options.filter((o) => !selectedOptions.includes(o));
+
+    setSeparatedOptions([...selectedOptions, ...unselectedOptions]);
+  };
+
+  useEffect(() => {
+    const invalid = () => setError(true);
+
+    inputRef.current?.addEventListener("invalid", invalid);
+    return () => {
+      inputRef.current?.removeEventListener("invalid", invalid);
+    };
+  }, [inputRef]);
+
+  useEffect(() => {
+    processValue(value);
+  }, [value]);
+
+  useEffect(() => {
+    sortOptions();
+  }, [options, val]);
+
+  return (
+    <Grid md={gridWidth || 6} xs={12} item>
+      <StyledFormControl fullWidth error={error}>
+        <StyledLabel htmlFor={id} id={`${id}-label`}>
+          {label}
+          {required ? <StyledAsterisk /> : ""}
+          {tooltipText && <Tooltip placement="right" title={tooltipText} />}
+        </StyledLabel>
+        <StyledAutocomplete
+          value={val}
+          onChange={onChangeWrapper}
+          options={separatedOptions}
+          readOnly={readOnly}
+          getOptionLabel={(option: string) => option}
+          renderTags={(value: string[]) => {
+            if (value?.length === 0 || hasFocus) {
+              return null;
+            }
+
+            if (value.length === 1) {
+              return value[0];
+            }
+
+            return tagText(value);
+          }}
+          forcePopupIcon
+          popupIcon={<DropdownArrowsIconSvg />}
+          slotProps={{ popper: { disablePortal: true } }}
+          renderInput={(params) => (
+            <TextField
+              {...params}
+              inputRef={inputRef}
+              placeholder={val?.length > 0 ? undefined : placeholder}
+              id={id}
+              onFocus={() => !readOnly && setHasFocus(true)}
+              onBlur={handleInputBlur}
+              onKeyDown={(event) => {
+                // Prevent backspace from clearing input tags
+                if (event.key === "Backspace") {
+                  event.stopPropagation();
+                }
+              }}
+            />
+          )}
+          {...rest}
+        />
+        <StyledHelperText>{!readOnly && error ? helperText : " "}</StyledHelperText>
+      </StyledFormControl>
+      <ProxySelect name={name} aria-labelledby={`${id}-label`} multiple hidden>
+        {val.map((v) => (
+          <option key={v} value={v} aria-label={v} selected />
```

Comment:
This "selected" prop is causing an error, see below.

![image](https://github.com/user-attachments/assets/11d880c8-f595-46de-83a7-5ca19aee553e)


I quickly grabbed this from the proxy select for the SelectInput component also used in the questionnaire, but it seems to work the same without the error. Feel free to change it how you see fit:

```javascript
<ProxySelect
  value={val}
  onChange={() => {}}
  name={name}
  aria-labelledby={`${id}-label`}
  multiple
  hidden
  >
    {val.map((v) => (
      <option key={v} value={v} aria-label={v} />
    ))}
</ProxySelect>
```

Reply:
Thanks for catching this, I always forget to check the console. Updated in 01753b7